proposals/20210723-feature_selection_custom_component.md

```diff
@@ -0,0 +1,37 @@
+#### SIG TFX-Addons
+# Project Proposal
+
+**Your name:** Nirzari Gupta
+
+**Your email:** nirzu97@gmail.com
+
+**Your company/organization:** Outreachy
+
+**Project name:** [Feature selection custom component](https://github.com/tensorflow/tfx-addons/issues/7)
+
+## Project Description
+This project provides a facility to perform various feature selection algorithms on datasets in TFX pipelines. Additionally, feature scores for selected features will also be generated as a custom artifact.
+
+## Project Category
+Component
+
+## Project Use-Case(s)
+This project will allow the user to select different algorithms for performing feature selection on datasets artifacts in TFX pipelines.
+
+## Project Implementation
+Feature Selection Custom Component will be implemented as Python function-based component.
+Implementation of the Feature Selection Custom Component can be done using the following approach:
+- Get dataset artifact generated by ExampleGen
+- Convert it into the format compatible with Scikit-Learn functions

```

Comment:
Would be good to discuss somewhere potential pitfalls of scalability of this approach (and also surface if possible alternative approaches). It may be okay here to run sklearn on a subsample of the dataset, but worried this should be documented.